generated/nidigital/nidigital/session.py

```diff
@@ -1113,73 +1437,194 @@ def get_fail_count(self):
-        name_buffer_size_ctype = _visatype.ViInt32()  # case S170
-        name_ctype = None  # case C050
-        error_code = self._library.niDigital_GetPinName(vi_ctype, pin_index_ctype, name_buffer_size_ctype, name_ctype)
+        site_list_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010
+        site_result_type_ctype = _visatype.ViInt32(site_result_type.value)  # case S130
+        site_numbers_buffer_size_ctype = _visatype.ViInt32(0)  # case S190
+        site_numbers_ctype = None  # case B610
+        actual_num_site_numbers_ctype = _visatype.ViInt32()  # case S220
+        error_code = self._library.niDigital_GetSiteResultsSiteNumbers(vi_ctype, site_list_ctype, site_result_type_ctype, site_numbers_buffer_size_ctype, site_numbers_ctype, None if actual_num_site_numbers_ctype is None else (ctypes.pointer(actual_num_site_numbers_ctype)))
         errors.handle_error(self, error_code, ignore_warnings=True, is_error_handling=False)
-        name_buffer_size_ctype = _visatype.ViInt32(error_code)  # case S180
-        name_ctype = (_visatype.ViChar * name_buffer_size_ctype.value)()  # case C060
-        error_code = self._library.niDigital_GetPinName(vi_ctype, pin_index_ctype, name_buffer_size_ctype, name_ctype)
+        site_numbers_buffer_size_ctype = _visatype.ViInt32(actual_num_site_numbers_ctype.value)  # case S200
+        site_numbers_size = actual_num_site_numbers_ctype.value  # case B620
+        site_numbers_ctype = get_ctypes_pointer_for_buffer(library_type=_visatype.ViInt32, size=site_numbers_size)  # case B620
+        error_code = self._library.niDigital_GetSiteResultsSiteNumbers(vi_ctype, site_list_ctype, site_result_type_ctype, site_numbers_buffer_size_ctype, site_numbers_ctype, None if actual_num_site_numbers_ctype is None else (ctypes.pointer(actual_num_site_numbers_ctype)))
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
-        return name_ctype.value.decode(self._encoding)
+        return [int(site_numbers_ctype[i]) for i in range(site_numbers_buffer_size_ctype.value)]
 
     @ivi_synchronized
-    def _get_pin_results_pin_information(self):
-        r'''_get_pin_results_pin_information
+    def is_site_enabled(self):
+        r'''is_site_enabled
 
         TBD
 
         Tip:
-        This method requires repeated capabilities (channels). If called directly on the
+        This method requires repeated capabilities (sites). If called directly on the
         nidigital.Session object, then the method will use all repeated capabilities in the session.
         You can specify a subset of repeated capabilities using the Python index notation on an
         nidigital.Session repeated capabilities container, and calling this method on the result.:
 
-            session.channels[0,1]._get_pin_results_pin_information()
+            session.sites[0,1].is_site_enabled()
```

Comment:
This snippet is hardcoded in the generator. See `documentation_snippets.py`. At this point the generator doesn't have a concept of rep caps that are single item vs list.

Adding this would be a somewhat bigger change to the code generator as well as expanding the metadata to know this information.

Reply:
Added this to an existing [issue ](https://github.com/ni/nimi-python/issues/1125#issuecomment-594638605) about making repeated capabilities docstring better.